Assets/Mirror/Runtime/NetworkManager.cs

```diff
@@ -938,7 +938,9 @@ void FinishLoadSceneHost()
 
             if (clientReadyConnection != null)
             {
+#pragma warning disable 618
                 OnClientConnect(clientReadyConnection);
```

Comment:
what about the new function?
ideally we would call:

# pragma...
OldFunction()
# end pragma
NewFunction()

so that both work.

Reply:
We're calling the old one and suppressing the warning because it's virtual.  That calls the new one by default, or if overridden and the base of the override is called.

```cs
        [Obsolete("Remove the NetworkConnection parameter in your override and use NetworkClient.connection instead.")]
        public virtual void OnClientConnect(NetworkConnection conn) => OnClientConnect();
```

Reply:
I noticed you already do 

#pragma
old()
#end
new()

above in lobby manager etc.
it would be better to do this everywhere else too because some day we'll remove the old one, and might accidentally remove the new call too.

where as with 
#pragma
old()
#end
new()

it's 100% obvious that we won't remove the new one

Reply:
Calling both would not make sense.  pragma isn't a compiler define.  If I've done that anywhere I'll fix it.

Reply:
Three months from now the obsoletes will be removed and that will force fixing the callers inside the pragmas to the correct method.

Reply:
@vis2k I can find no place I do this...it would be erroneous.
```cs
#pragma
old()
#end
new()
```

Reply:
here is what I mean.
let's say we have a callback OnCallback(param).
we want to replace it with OnCallback() without param.

for the end user, both should work.
he can keep using OnCallback(param).
or if he wants to, he can move to OnCallback().

Mirror previously called OnCallback(param).
Now it should simply call:
OnCallback(param);
OnCallback();
so that both are supported.

that's how the obsolete interest management system works, too.
we call both the obsolete function and the new function:
<img width="739" alt="2021-07-11_15-31-58@2x" src="https://user-images.githubusercontent.com/16416509/125186580-a8659e80-e25d-11eb-898f-dba5ed203b97.png">

Reply:
You do realize that all the obsolete methods call the new methods, right?

Reply:
> **for the end user**, both should work.
> he can keep using obsolete OnCallback(param).
> or if he wants to, he can move to OnCallback().

This is how I have it working...really.  They'll get the obsolete warning but it does still work exactly as before.

Reply:
> You do realize that all the obsolete methods call the new methods, right?

yes, instead of making the obsoletes (which will be removed) call the new methods, we should call the new methods directly.

you already do it this way above in the room/match manager anyway

Reply:
I think you're confusing code in an example with Mirror's core code.  We've had the obsoletes call the new methods many many times.  I'm sure there's other such cases in Mirror today.  As I said, in 3 months or whatever when the obsoletes are removed, the calls to them will be updated to the new ones.